lntemp/rpc/lnd.go

```diff
@@ -0,0 +1,206 @@
+	defer cancel()
+
+	info, err := h.LN.GetInfo(ctxt, &lnrpc.GetInfoRequest{})
+	require.NoErrorf(h, err, "failed to GetInfo for node: %s", h.Name)
+
+	return info
+}
+
+// ConnectPeer makes a RPC call to ConnectPeer and asserts there's no error.
+func (h *HarnessRPC) ConnectPeer(
+	req *lnrpc.ConnectPeerRequest) *lnrpc.ConnectPeerResponse {
+
+	ctxt, cancel := context.WithTimeout(h.runCtx, DefaultTimeout)
+	defer cancel()
+
+	resp, err := h.LN.ConnectPeer(ctxt, req)
+	require.NoErrorf(h, err, "failed to ConnectPeer for node: %s", h.Name)
+	return resp
+}
+
+// ListChannels list the channels for the given node and asserts it's
+// successful.
+func (h *HarnessRPC) ListChannels(
+	req *lnrpc.ListChannelsRequest) *lnrpc.ListChannelsResponse {
+
+	ctxt, cancel := context.WithTimeout(h.runCtx, DefaultTimeout)
+	defer cancel()
+
+	resp, err := h.LN.ListChannels(ctxt, req)
+	require.NoErrorf(h, err, "unable ListChannels for %s", h.Name)
+
+	return resp
+}
+
+// PendingChannels makes a RPC request to PendingChannels and asserts there's
+// no error.
+func (h *HarnessRPC) PendingChannels() *lnrpc.PendingChannelsResponse {
+	ctxt, cancel := context.WithTimeout(h.runCtx, DefaultTimeout)
+	defer cancel()
+
+	pendingChansRequest := &lnrpc.PendingChannelsRequest{}
+	resp, err := h.LN.PendingChannels(ctxt, pendingChansRequest)
+
+	require.NoErrorf(h, err, "failed to PendingChannels for %s", h.Name)
+
+	return resp
+}
+
+// ClosedChannels makes a RPC call to node's ClosedChannels and asserts.
+func (h *HarnessRPC) ClosedChannels(
+	req *lnrpc.ClosedChannelsRequest) *lnrpc.ClosedChannelsResponse {
+
+	ctxt, cancel := context.WithTimeout(h.runCtx, DefaultTimeout)
+	defer cancel()
+
+	resp, err := h.LN.ClosedChannels(ctxt, req)
+	require.NoErrorf(h, err, "%s: list closed channels failed", h.Name)
+
+	return resp
+}
+
+// ListPayments lists the node's payments and asserts.
+func (h *HarnessRPC) ListPayments(
+	req *lnrpc.ListPaymentsRequest) *lnrpc.ListPaymentsResponse {
+
+	ctxt, cancel := context.WithTimeout(h.runCtx, DefaultTimeout)
+	defer cancel()
+
+	resp, err := h.LN.ListPayments(ctxt, req)
+	require.NoErrorf(h, err, "%s: failed to list payments", h.Name)
+
+	return resp
+}
+
+// ListInvoices list the node's invoice using the request and asserts.
+func (h *HarnessRPC) ListInvoices(
+	req *lnrpc.ListInvoiceRequest) *lnrpc.ListInvoiceResponse {
+
+	ctxt, cancel := context.WithTimeout(h.runCtx, DefaultTimeout)
+	defer cancel()
+
+	resp, err := h.LN.ListInvoices(ctxt, req)
+	require.NoErrorf(h, err, "%s: list invoice failed", h.Name)
+
+	return resp
+}
+
+// ListUnspent makes a RPC call to ListUnspent and asserts.
+func (h *HarnessRPC) ListUnspent(
+	req *lnrpc.ListUnspentRequest) *lnrpc.ListUnspentResponse {
+
+	ctxt, cancel := context.WithTimeout(h.runCtx, DefaultTimeout)
+	defer cancel()
+
+	resp, err := h.LN.ListUnspent(ctxt, req)
+	require.NoErrorf(h, err, "failed to list utxo for node %s", h.Name)
```

Comment:
Looks like we have 3 pattern for the error log:
 - `%s: <operation> failed`
 - `failed to <operation> for node: %s`
 - `failed to <operation> for %s`

Should we try to consolidate that? Perhaps we could even add a small helper method, for example:
```
func (h *HarnessRPC) NoError(err error, operation string) {
        require.NoErrorf(h, err, "%s: failed to %s", h.Name, operation)
}
```

And just call it as:
```
h.NoError(err, "list utxo")
```